en/dev_log/ulog_file_format.md

```diff
@@ -118,34 +114,34 @@ struct ulog_message_flag_bits_s {
 ```
 
 - `compat_flags`: compatible flag bits
-  - These flags indicate any information that is compatible with the existing ULog is in the log.
+  - These flags indicate whether any information that is compatible with the ULog file format is in the log.
   - `compat_flags[0]`: *DEFAULT_PARAMETERS* (Bit 0): if set, the log contains [default parameters message](#q-default-parameter-message)
 
   The rest of the bits are currently not defined and must be set to 0.
-  These bits can be used for future ULog changes that are compatible with existing parsers.
+  These bits can be used for future ULog changes that are compatible with existing parsers. For example, adding a new message types, which can be indicated by setting one of the Bits, and the existing parsers can just ignore the new message type in the log.
   It means parsers can just ignore the bits if one of the unknown bits is set.
 - `incompat_flags`: incompatible flag bits.
   - `incompat_flags[0]`: *DATA_APPENDED* (Bit 0): if set, the log contains appended data and at least one of the `appended_offsets` is non-zero.
   
-  All other bits are undefined and must be set to 0.
-  If a parser finds one of these bits set, it must refuse to parse the log.
-  This can be used to introduce breaking changes that existing parsers cannot handle.
-- `appended_offsets`: File offsets (0-based) for appended data.
+  The rest of the bits are currently not defined and must be set to 0.
+  This can be used to introduce breaking changes that existing parsers cannot handle. For example, when an old ULog Parser that didn't have the concept of *DATA_APPENDED* reads the newer ULog, it would stop parsing the log as the log will contain out-of-spec messages / concepts.
+  If a parser finds any of these bits set that isn't specified, it must refuse to parse the log.
+- `appended_offsets`: File offset for appended data.
   If no data is appended, all offsets must be zero.
-  This can be used to reliably append data for logs that may stop in the middle of a message.
+  This can be used to reliably append data for logs that may stop in the middle of a message. For example, hardfault dumps in PX4.
 
   A process appending data should do:
-  - set the relevant `incompat_flags` bit,
-  - set the first `appended_offsets` that is 0 to the length of the log file,
-  - then append any type of messages that are valid for the Data section.
+  - set the relevant `incompat_flags` bit
+  - set the first `appended_offsets` as the length of the log file without the appended data, as that is where the data would start
+  - append any type of messages that are valid for the Data section.
```

Comment:
I am still unsure why it's only the 'first' element of the array `appended_offsets` that is used. Why do we have 3 of them as an array then?

Also, I would like to check if the phrase "length of the log file without the appended data, as that is where the data would start" is correct.

In PX4, it seems that the `incompat_flags` isn't set anywhere, except for having a check in reply module for replaying the log. So where is this `incompat_flags` actually set? In PX4? Or external source? @bkueng